osschallenge/views.py

```diff
@@ -112,34 +120,25 @@ def MyTaskIndexView(request, username):
     template_name = 'osschallenge/mytasksindex.html'
     current_user_id = request.user.id
     user_task_objects = Task.objects.filter(assignee_id=current_user_id)
-    user_task_list = []
-    for obj in user_task_objects:
-        user_task_list.append(obj)
-    for task in user_task_objects:
+    try:
+        matches = user_task_objects.filter(
+            Q(title__icontains=request.GET.get('search')) |
+            Q(project__title__icontains=request.GET.get('search'))
+        ).distinct().order_by('id')
+    except ValueError:
+        matches = user_task_objects.order_by('id')
+    if request.GET.get('page'):
+        current_page = request.GET.get('page')
+    else:
+        current_page = 1
+    tasks, last_page, current_page = paging(current_page, matches, 5)
+    for task in tasks.object_list:
         task.description = shorten(task.description, max_length_description)
         task.title = shorten(task.title, max_length_title)
```

Comment:
This should be done in the template. Do you have a `shorten` template helper?

Reply:
I created a template filter for the shorten function.